server/src/main/java/com/flipkart/varadhi/web/AuthorizationHandlerBuilder.java

```diff
@@ -23,18 +25,28 @@
 public class AuthorizationHandlerBuilder {
 
     private final AuthorizationProvider provider;
+    private final Timer timer;
 
-    public AuthorizationHandlerBuilder(AuthorizationProvider provider) {
+    private final MeterRegistry meterRegistry;
+
+    public AuthorizationHandlerBuilder(AuthorizationProvider provider, MeterRegistry meterRegistry) {
         this.provider = Objects.requireNonNull(provider, "Authorization Provider is null");
+        this.meterRegistry = Objects.requireNonNull(meterRegistry, "Meter registry is null");
+        this.timer = Timer.builder("authorization.timer")
+                          .description("Time taken to check user authorization")
+                          .tag("method", "isAuthorized")
+                          .register(this.meterRegistry);
     }
 
     public AuthorizationHandler build(ResourceAction authorizationOnResource) {
         return new AuthorizationHandler(authorizationOnResource);
     }
 
     private Future<Void> authorizedInternal(UserContext userContext, ResourceAction action, String resourcePath) {
+        Timer.Sample clock = Timer.start(meterRegistry);
```

Comment:
you might want to create a simple utility in 
FutureExtensions class.

which will allow you to do this :
```java
return provider.isAuthorized(userContext, action, resourcePath).compose(authorized -> {
}).timed(timer);
```